lib/faraday/utils.rb

```diff
@@ -95,7 +95,14 @@ def to_hash() ::Hash.new.update(self) end
 
       def parse(header_string)
         return unless header_string && !header_string.empty?
-        header_string.split(/\r\n/).
+
+        headers = header_string.split(/\r\n/)
+
+        # Find the last set of response headers.
+        start_index = headers.rindex { |x| x.match(/^HTTP\//) }
```

Comment:
I see, I understand what you mean now.
I'll have a look at `rindex` implementation, I think what we're trying to add here is a fallback value in case there's no "HTTP" header. The maybe `.to_i` is not the best solution for that, but I get your point now!